kroxylicious/src/main/java/io/kroxylicious/proxy/config/tls/KeyStore.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.config.tls;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.net.ssl.KeyManagerFactory;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.netty.handler.ssl.SslContextBuilder;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+/**
+ * A {@link KeyProvider} backed by a Java Truststore.
+ *
+ * @param storeFile             location of a key store, or reference to a PEM file containing both private-key/certificate/intermediates.
+ * @param storePasswordProvider password used to protect the key store. cannot be used if trustType is PEM.
```

Comment:
Nit:
```suggestion
 * @param storePasswordProvider password used to protect the key store. Will be ignored if trustType is PEM.
```

I'm guessing thats more accurate as I don't think we throw exceptions if we find a pem file and a non null store password provider.